tests/Feature/EmailAssignmentTest.php

```diff
@@ -0,0 +1,176 @@
+            ->assertSessionHas('error', 'Emails entered are different.  You need to enter the same email, twice.');
+    }
+
+    /** @test * */
+    public function testSuccessfulPostEmail()
+    {
+        $data = [
+            'new_email' => 'email@example.com',
+            'new_email2' => 'email@example.com',
+        ];
+
+        $this->actingAs($this->account)->post(route('mship.manage.email.add.post'), $data)
+            ->assertRedirect(route('mship.manage.dashboard'))
+            ->assertSessionHas('success');
+    }
+
+    /** @test * */
+    public function testDuplicateEmailOnPost()
+    {
+        $account = $this->account->secondaryEmails()->create(['email' => 'email2@example.com']);
+
+        $data = [
+            'new_email' => 'email2@example.com',
+            'new_email2' => 'email2@example.com',
+        ];
+
+        $this->actingAs($this->account->fresh())->post(route('mship.manage.email.add.post'), $data)
+            ->assertRedirect(route('mship.manage.dashboard'))
+            ->assertSessionHas('error', 'This email has already been added to your account.');
+    }
+
+    /** @test * */
+    public function testRedirectOnSecondaryEmailDeleted()
+    {
+        $account = $this->account->secondaryEmails()->create(['email' => 'secondary.email@example.com']);
+
+        $data = [
+            'id' => $account->id,
+        ];
+
+        $this->actingAs($this->account->fresh())->post(route('mship.manage.email.delete.post', $account), $data)
+            ->assertRedirect(route('mship.manage.dashboard'))
+            ->assertSessionHas('success', 'Your secondary email ('.$account->email.') has been removed!');
+    }
+
+    /** @test * */
+    public function testSuccessfulSecondaryEmailAddViaGetHasRelevantData()
+    {
+        $account = $this->account->secondaryEmails()->create(['email' => 'secondary.email@example.com']);
+
+        $data = [
+            'id' => $account->id,
+        ];
+
+        $this->actingAs($this->account->fresh())->get(route('mship.manage.email.delete.post', $account), $data)
+            ->assertViewIs('mship.management.email.delete')
+            ->assertViewHas(['email' => $account->email]);
+    }
+
+    /** @test * */
+    public function testAssignmentsEmailsPassedToView()
+    {
+        $account = $this->account->secondaryEmails()->create(['email' => 'secondary.email@example.com']);
+
+        $email = $this->account->fresh()->email;
+
+        $this->actingAs($this->account)->get(route('mship.manage.email.assignments'))
+            ->assertViewIs('mship.management.email.assignments')
+            ->assertViewHas([
+                'userPrimaryEmail' => $email,
+                'userSecondaryVerified' => $this->account->verified_secondary_emails,
+            ]);
+    }
+
+    /** @test * */
+    public function testUserCannotDeleteOtherUsersEmail()
+    {
+        $otherEmail = 'email@otheruser.co.uk';
+
+        $accountOther = factory(\App\Models\Mship\Account::class)->create();
```

Comment:
Can you refactor this (`$otherEmail` and `$accountOther`) into the setup function to avoid the duplication on lines 143 too?